### PR TITLE
Update projects 1.4, 1.5 and 2.1. Update the mentor for project 3.2.

### PR DESCRIPTION
**Pull request by seanlip**

### CODE REVIEW

**Comment by seanlip** [comment]
Adding @nikitaevg for review of changes in 1.5.

/cc @Nik-09 @chris7716 @vojtechjelinek as FYI since there are small changes to your project descriptions, thanks!

File: Google-Summer-of-Code-2024.md [hunk 1]
```diff
@@ -650,15 +652,13 @@ Link to PRD: https://docs.google.com/document/d/1r9IEQ5z_t-eu9XAWN3eRA7iKdKuYsOQ
 Note: Ability to [write Beam jobs](https://github.com/oppia/oppia/wiki/Apache-Beam-Jobs) is a plus, but is not strictly required.
 
 **Suggested Milestones:**
-- **Milestone 1**: Add a classification level for each skill as trivial/simple/regular, and allow the curriculum admin / topic manager to update this classification in the skill editor (with a default to regular). Include help text for them to explain what each classification means.
+- **Milestone 1**: Track each user’s total points (this should be updated in the backend each time a learner answers a question correctly, regardless of whether they eventually complete the practice session or not), and show these totals in their learner dashboard. The tracking functionality should be hidden behind a `TRACK_PRACTICE_QUESTION_POINTS` feature flag, and the display functionality should be hidden behind a `SHOW_PRACTICE_QUESTION_POINTS` feature flag.
```
Comment (nikitaevg) [inline Google-Summer-of-Code-2024.md]: > and show these totals

and show these points?

File: Google-Summer-of-Code-2024.md [hunk 2]
```diff
@@ -650,15 +652,13 @@ Link to PRD: https://docs.google.com/document/d/1r9IEQ5z_t-eu9XAWN3eRA7iKdKuYsOQ
...
+  Implement the redesigned practice sessions UI behind a `SHOW_UPDATED_PRACTICE_SESSIONS_UI` flag, including counters and animations for showing how many points a user has earned in a practice session.
 
-  Track each user’s total points/stars (this should be updated on completion of each practice session), and show these totals in their learner dashboard. The tracking functionality should be hidden behind a `TRACK_POINTS_AND_STARS` feature flag, and the display functionality should be hidden behind a `SHOW_POINTS_AND_STARS` feature flag.
+  Add functionality for tracking mastery at both the skill and subtopic levels (this includes storing information about "number correct", "total number attempted", and "results of last 100 questions attempted"). Use this to track answers for questions that are done in both lessons and practice sessions, and update the learner’s mastery scores in the backend. This functionality should be hidden behind a `TRACK_SKILL_MASTERY` feature flag.
```
Comment (nikitaevg) [inline Google-Summer-of-Code-2024.md:659]: skill and subtopic -> skill and concept?

**Review by nikitaevg — APPROVED** [review]
Left a couple comments, thanks!

File: Google-Summer-of-Code-2024.md (hunk 1 above)
Comment (seanlip) [reply Google-Summer-of-Code-2024.md]: Should probably be "and show this total". Will fix, thanks!

File: Google-Summer-of-Code-2024.md (hunk 2 above)
Comment (seanlip) [reply Google-Summer-of-Code-2024.md:659]: No, subtopic is a group of skills, and skill is the same as concept. (At least currently until we implement any renamings, but UX writing team still needs to test these.)

**Review by chris7716 — APPROVED** [review]
LGTM!

File: Google-Summer-of-Code-2024.md (hunk 1 above)
Comment (seanlip) [reply Google-Summer-of-Code-2024.md]: Done

**Pull request merged**